src/main/kotlin/interactor/GetCityThatMatchTheManagerExpectationsInteractor.kt

```diff
@@ -1,61 +1,40 @@
 package interactor
 
 import model.CityEntity
-import java.util.*
+import kotlin.math.abs
 
 
 class GetCityThatMatchTheManagerExpectationsInteractor(
-    private val dataSource: CostOfLivingDataSource
+    private val dataSource: CostOfLivingDataSource,
 ) {
 
-    fun execute(countries: List<String>): CityEntity? {
-        val existCountries = countries.map(::formatCountriesNames)
-            .filter { selectedCountries.contains(it) }
-
-        val pricesOfMeals = mutableListOf<Float>()
-
-        return if (existCountries.isNotEmpty()) {
-            dataSource
-                .getAllCitiesData()
-                .filter { it.country in existCountries }
-                .filter(::excludeNullPricesOfMeals)
-                .onEach {
-                    pricesOfMeals.add(it.mealsPrices.mealFor2PeopleMidRangeRestaurant!!)
-                }.let { citiesEntity ->
-                    getCityThatMatchExpectations(pricesOfMeals, citiesEntity)
-                }
-        } else {
-            null
-        }
-    }
-
+    fun execute(selectedCountries: List<String>): CityEntity {
+        val countries = selectedCountries.map { it.lowercase() }
 
-    private fun formatCountriesNames(countryName: String): String {
-        return countryName.lowercase()
-            .trim()
-            .split("\\s+".toRegex())
-            .joinToString(" ") {
-                it.replaceFirstChar { char ->
-                    if (char.isLowerCase()) char.titlecase(Locale.getDefault()) else it
-                }
-            }
+        return dataSource.getAllCitiesData()
+            .filter { countries.contains(it.country.lowercase())  }
+            .takeIf { it.isNotEmpty() }
+            ?.run {
+                excludeNullPricesOfMeals()
+               .getCityThatMatchExpectations()
+            } ?: throw IllegalStateException("List of cities is empty")
     }
 
-    private fun excludeNullPricesOfMeals(city: CityEntity): Boolean {
-        return city.mealsPrices.mealFor2PeopleMidRangeRestaurant != null
+    private fun List<CityEntity>.excludeNullPricesOfMeals() = filter {
```

Comment:
here not need extension function can put filter in run scope and move block this function into model "MealsPrices" .

Reply:
not use **compact function** same this case.